library/src/main/java/com/jwoolston/android/uvc/interfaces/streaming/VideoFrame.java

```diff
@@ -27,125 +27,138 @@
-                | ((0xFF & descriptor[dwDefaultFrameInterval + 2]) << 8) | (0xFF & descriptor[dwDefaultFrameInterval + 3]);
-
-        if (mFrameIntervalType == 0) {
-            mMinFrameInterval = ((0xFF & descriptor[dwMinFrameInterval]) << 24) | ((0xFF & descriptor[dwMinFrameInterval + 1]) << 16)
-                    | ((0xFF & descriptor[dwMinFrameInterval + 2]) << 8) | (0xFF & descriptor[dwMinFrameInterval + 3]);
-            mMaxFrameInterval = ((0xFF & descriptor[dwMaxFrameInterval]) << 24) | ((0xFF & descriptor[dwMaxFrameInterval + 1]) << 16)
-                    | ((0xFF & descriptor[dwMaxFrameInterval + 2]) << 8) | (0xFF & descriptor[dwMaxFrameInterval + 3]);
-            mFrameIntervalStep = ((0xFF & descriptor[dwFrameIntervalStep]) << 24) | ((0xFF & descriptor[dwFrameIntervalStep + 1]) << 16)
-                    | ((0xFF & descriptor[dwFrameIntervalStep + 2]) << 8) | (0xFF & descriptor[dwFrameIntervalStep + 3]);
-            mFrameIntervals = null;
+        frameIntervalType = (0xFF & descriptor[bFrameIntervalType]);
+        if (frameIntervalType == 0 && descriptor.length < LENGTH_INTERVAL_TYPE_0) {
+            throw new IllegalArgumentException(
+                    "The provided descriptor is not long enough to be an Uncompressed Video Frame.");
+        }
+        if (frameIntervalType != 0 && descriptor.length < (MIN_LENGTH_INTERVAL_TYPE_NOT_0 + 4 * frameIntervalType)) {
+            throw new IllegalArgumentException(
+                    "The provided descriptor is not long enough to be an Uncompressed Video Frame.");
+        }
+        frameIndex = (0xFF & descriptor[bFrameIndex]);
+        stillImageSupported = (descriptor[bmCapabilites] & 0x01) != 0;
+        fixedFrameRateEnabled = (descriptor[bmCapabilites] & 0x02) != 0;
+
+        width = ((0xFF & descriptor[wWidth + 1]) << 8) | (0xFF & descriptor[wWidth]);
+        height = ((0xFF & descriptor[wHeight + 1]) << 8) | (0xFF & descriptor[wHeight]);
+
+        minBitRate = ((0xFF & descriptor[dwMinBitRate + 3]) << 24) | ((0xFF & descriptor[dwMinBitRate + 2]) << 16)
+                     | ((0xFF & descriptor[dwMinBitRate + 1]) << 8) | (0xFF & descriptor[dwMinBitRate]);
+        maxBitRate = ((0xFF & descriptor[dwMaxBitRate + 3]) << 24) | ((0xFF & descriptor[dwMaxBitRate + 2]) << 16)
+                     | ((0xFF & descriptor[dwMaxBitRate + 1]) << 8) | (0xFF & descriptor[dwMaxBitRate]);
+        defaultFrameInterval = ((0xFF & descriptor[dwDefaultFrameInterval + 3]) << 24)
+                               | ((0xFF & descriptor[dwDefaultFrameInterval + 2]) << 16)
+                               | ((0xFF & descriptor[dwDefaultFrameInterval + 1]) << 8)
+                               | (0xFF & descriptor[dwDefaultFrameInterval]);
+
+        if (frameIntervalType == 0) {
+            minFrameInterval = ((0xFF & descriptor[dwMinFrameInterval + 3]) << 24)
```

Comment:
Looks like the same thing is being done in multiple places here which appears to be int extractions. Should probably pull these out into a few static methods.